File: android/src/main/java/com/pichillilorenzo/flutter_inappwebview/plugin_scripts_js/PrintJS.java
package com.pichillilorenzo.flutter_inappwebview.plugin_scripts_js;

import com.pichillilorenzo.flutter_inappwebview.types.PluginScript;
import com.pichillilorenzo.flutter_inappwebview.types.UserScriptInjectionTime;

public class PrintJS {
  public static final String PRINT_JS_PLUGIN_SCRIPT_GROUP_NAME = "IN_APP_WEBVIEW_PRINT_JS_PLUGIN_SCRIPT";
  public static final PluginScript PRINT_JS_PLUGIN_SCRIPT = new PluginScript(
          PrintJS.PRINT_JS_PLUGIN_SCRIPT_GROUP_NAME,
          PrintJS.PRINT_JS_SOURCE,
          UserScriptInjectionTime.AT_DOCUMENT_START,
          null,
          false
  );

  public static final String PRINT_JS_SOURCE = "window.print = function() {" +
          "  if (window.top == null || window.top === window) {" +
          "     window." + JavaScriptBridgeJS.JAVASCRIPT_BRIDGE_NAME + ".callHandler('onPrint', window.location.href);" +
          "  } else {" +
          "     window.top.print();" +
          "  }" +
          "};";
}

File: flutter_inappwebview_ios/ios/Classes/PluginScriptsJS/PrintJS.swift
//
//  PrintJS.swift
//  flutter_inappwebview
//
//  Created by Lorenzo Pichilli on 16/02/21.
//

import Foundation

public class PrintJS {
    
    public static let PRINT_JS_PLUGIN_SCRIPT_GROUP_NAME = "IN_APP_WEBVIEW_PRINT_JS_PLUGIN_SCRIPT"
    
    public static func  PRINT_JS_PLUGIN_SCRIPT(allowedOriginRules: [String]?, forMainFrameOnly: Bool) -> PluginScript {
        return PluginScript(
            groupName: PRINT_JS_PLUGIN_SCRIPT_GROUP_NAME,
            source: PRINT_JS_SOURCE(),
            injectionTime: .atDocumentStart,
            forMainFrameOnly: forMainFrameOnly,
            allowedOriginRules: allowedOriginRules,
            requiredInAllContentWorlds: true,
            messageHandlerNames: [])
    }
    
    public static func PRINT_JS_SOURCE() -> String {
        return """
        window.print = function() {
            window.\(JavaScriptBridgeJS.get_JAVASCRIPT_BRIDGE_NAME()).callHandler("onPrintRequest", window.location.href);
        }
        """
    }
}
